Merge Carchet TPMS with EezTire TPMS

kesuhm opened the pull request:
I merged the code for these two sensors because they appear to use the same protocol. I added low battery and inflate flags to the readout.  The low battery flag has been verified with the Carchet dashboard monitor, but not the EezTire monitor. During low battery flag testing, I was able to trigger the second flags byte (flags2) to a value of 0x20, but was unable to controllably reproduce this.

I also commented in operations which can be used to display pressure in PSI and temperature in deg F.

zuckschwerdt commented:
Thanks. Very good improvements!
- "8 bit little-endian"... only multi-byte can have endianess, right?
- "int temperature_C = b[4] - 50;//int temperature_F = b[4];" can you check with extreme temperatures if it's F or C?
- "//float pressure_PSI = b[6] * 0.363 - 0.06946;" that sounds like 2.5 kPa steps (nice round number)
- "low batt" must be "battery_ok", see other decoders.
- "EezTire E618 | Carchet" this should be "EezTire E618, Carchet TPMS"

zuckschwerdt commented:
Please stick with a PR. Don't just abandon to open new ones.
Note: this continues #2677

kesuhm commented:
I'm sorry. I didn't realize the green button would assign a new PR number every time I pushed it. I am now working back in #2677 -- I have merged the Carchet TPMS code with the EezTire code, corrected my terminology, and fixed my Fahrenheit math comment. In my first merge, I had missed the unit conversion I had done in the Carchet TPMS data_make code.

zuckschwerdt commented:
The code is manually merged with a7e5aaf
